libcudacxx/include/cuda/std/detail/libcxx/include/__cuda/barrier.h

```diff
@@ -525,15 +525,15 @@ friend class _CUDA_VSTD::__barrier_poll_tester_parity;
 
     template<class _Rep, class _Period>
     _LIBCUDACXX_NODISCARD_ATTRIBUTE _LIBCUDACXX_INLINE_VISIBILITY
-    bool try_wait_for(arrival_token && __token, const _CUDA_VSTD::chrono::duration<_Rep, _Period>& __dur) {
+    bool try_wait_for(arrival_token & __token, const _CUDA_VSTD::chrono::duration<_Rep, _Period>& __dur) {
```

Comment:
Can we add a test that would have caught this? e.g., a `try_wait_for` that fails and retries.

Reply:
The current tests for this are a little janky: https://github.com/NVIDIA/cccl/blob/137f5c7cd843e12c819bb4cc6033e7bc3d23dc60/libcudacxx/.upstream-tests/test/std/thread/thread.barrier/try_wait_for.pass.cpp#L41

It's moving from a moved-from object in a loop.

Reply:
This may not actually be possible to do without adding some runtime state tracking to track a moved-from token. So it may not be worth it.

Reply:
Main thing is to just remove the `std::move` which would catch not needing to pass an r-value ref.